app/code/Magento/CatalogRule/Pricing/Price/CatalogRulePrice.php

```diff
@@ -99,9 +99,9 @@ public function getValue()
                         $this->product->getId()
                     );
                 $this->value = $this->value ? floatval($this->value) : false;
-                if ($this->value) {
-                    $this->value = $this->priceCurrency->convertAndRound($this->value);
-                }
+            }
+            if ($this->value) {
```

Comment:
```php
if ($this->product->hasData(self::PRICE_CODE)) {
    $this->value = floatval($this->product->getData(self::PRICE_CODE)) ?: false;
```
This value is stored in the 'catalogrule_product_price' table. As I know all prices in Magento is converted after (from base currency to selected on site) and it's too much work to store prepared value because there is should be added a new dimension by currency per store. Value which is stored in product is website-aware, but currency is another dimension (you can have many currencies for one store view).